include/execution.hpp

```diff
@@ -553,8 +586,14 @@ namespace std::execution {
             self.continuation_.resume();
           }
 
-          friend void tag_invoke(set_error_t, __rec_base&& self, exception_ptr eptr) noexcept {
-            self.result_->template emplace<2>((exception_ptr&&) eptr);
+          template <class Error>
+          friend void tag_invoke(set_error_t, __rec_base&& self, Error&& err) noexcept {
+            if constexpr (__decays_to<Error, exception_ptr>)
+              self.result_->template emplace<2>((Error&&) err);
+            else if constexpr (__decays_to<Error, error_code>)
+              self.result_->template emplace<2>(make_exception_ptr(system_error(err)));
+            else
+              self.result_->template emplace<2>(make_exception_ptr((Error&&) err));
```

Comment:
I think @kirkshoop had a preference to have a CPO / algorithm for converting an error to an exception_ptr rather than hard-coding logic like this here.

Reply:
I tend to agree, but it's something we can save for R4. I'm running out of time. It is also worth asking LEWG their opinion on the matter.